routers/web/repo/setting/setting.go

```diff
@@ -865,6 +857,43 @@ func SettingsPost(ctx *context.Context) {
 		ctx.Flash.Success(ctx.Tr("repo.settings.wiki_deletion_success"))
 		ctx.Redirect(ctx.Repo.RepoLink + "/settings")
 
+	case "change-visiblity":
+		if ctx.HasError() {
+			ctx.HTML(http.StatusOK, tplSettingsOptions)
+			return
+		}
+		if !ctx.Repo.IsOwner() {
+			ctx.Error(http.StatusNotFound)
+			return
+		}
```

Comment:
The old code doesn't have such check.

IIRC the `IsOwner` should have been correctly checked by a middleware for this handler.